stac_generator/raster_data/schema.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import annotations
+
+import json
+from datetime import UTC, datetime
+from typing import Any
+
+from pydantic import BaseModel, Field, model_validator
+
+from stac_generator.base.schema import SourceConfig
+
+
+class BandInfo(BaseModel):
+    """Band information for raster data"""
+
+    name: str
+    wavelength: float | None = Field(default=None)  # Can be float or None
+    nodata: float | None = Field(default=0)  # Default nodata value
+    data_type: str | None = Field(default="uint16")  # Default data type for raster band
+
+    @model_validator(mode="before")
+    @classmethod
+    def parse_wavelength(cls, data: Any) -> Any:
```

Comment:
I don't know whether this is necessary. If there is no wavelength information, they should just leave it None or blank instead of "no band specified". This should also just be a TypedDict since it doesn't need validation.